mozci/push.py

```diff
@@ -294,14 +296,36 @@ def add(result):
         except MissingDataError:
             pass
 
-        # Gather information from the unittest table. We allow missing data for this table because
-        # ActiveData only holds very recent data in it, but we have fallbacks on Taskcluster
-        # artifacts.
-        try:
-            add(run_query("push_tasks_results_from_unittest", args))
-        except MissingDataError:
-            pass
+        # Let's gather error/results from cache or AD/Taskcluster
+        test_tasks_results = adr.config.cache.get(self.push_uuid, {})
```

Comment:
I'm thinking an implementation like this would be cleaner and the diff would be much smaller:
```
test_tasks_results = adr.config.cache.get(self.push_uuid, {})
was_cached = len(test_tasks_results.keys()) != 0
if not was_cached:
    # get from AD

# normalize tasks

# "extract" test_tasks_results

# load errors and results for all tasks

if not was_cached:
   # store to cache
```

WDYT?

Reply:
Let me know what you think of the refactoring.